### PR TITLE
Feat/G0 5928: Monthly and Ledger columnar report 

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/exportLedger/exportLedger.component.html [hunk 1]
```diff
@@ -69,21 +68,26 @@ <h3>Export Ledger</h3>
...
             <div class="modal_wrap" [hidden]="emailTypeSelected === 'columnar'">
                 <div class="clearfix"></div>
                 <h3 class="pdT1 pdB1">Or</h3>
                 <div>
                     <h3 class="l8grey pdB size-xs">You can add multiple id's separated comma</h3>
-                    <textarea [(ngModel)]="emailData" required name="email" class="form-control"
-                        placeholder="Recipents Email Id's"></textarea>
-                    <button ng-disabled="ledgerEmailSendForm.$invalid" class="btn btn-success mrT1"
-                        (click)="sendLedgEmail()">Send
+                    <textarea [(ngModel)]="emailData" required name="email" class="form-control" placeholder="Recipents Email Id's"></textarea>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/exportLedger/exportLedger.component.html:84]: Recipents spelling is wrong

File: apps/web-giddh/src/app/ledger/components/exportLedger/exportLedger.component.ts [hunk 2]
```diff
@@ -57,12 +61,13 @@ export class ExportLedgerComponent implements OnInit {
...
+        exportRequest.balanceTypeAsSign = this.balanceTypeAsSign;
         const body = _.cloneDeep(this.advanceSearchRequest);
+        body.dataToSend.balanceTypeAsSign = this.balanceTypeAsSign;
         if (!body.dataToSend.bsRangeValue) {
-            this.universalDate$.pipe(take(1)).subscribe(a => {
-                if (a) {
-                    body.dataToSend.bsRangeValue = [moment(a[0], 'DD-MM-YYYY').toDate(), moment(a[1], 'DD-MM-YYYY').toDate()];
+            this.universalDate$.pipe(take(1)).subscribe(res => {
+                if (res) {
+                    body.dataToSend.bsRangeValue = [moment(res[0], 'DD-MM-YYYY').toDate(), moment(res[1], 'DD-MM-YYYY').toDate()];
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/exportLedger/exportLedger.component.ts:70]: Use date format constant

File: apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                            </th>
+                        </ng-container>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr *ngFor="let item of reportResponseResult; let i = index;">
+                        <td data-title="S. No" class="text-center">{{ i + 1 }} </td>
+                        <td data-title="Date" class="text-center">{{ item.date }} </td>
+                        <td data-title="Particulars" class="text-center">{{ item.baseAccount }} </td>
+                        <td data-title="S. No" class="text-center">{{ item.address }} </td>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.html:51]: Duplicate data-title. Correct all duplicated data-title

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 4]
```diff
@@ -1804,4 +1807,27 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
+     * @param {{ isShowColumnarTable: boolean, exportRequest: ExportLedgerRequest }} event Ccolumnar report emmiter event object
+     * @memberof LedgerComponent
+     */
+    public onShowColumnarReportTable(event: { isShowColumnarTable: boolean, exportRequest: ExportLedgerRequest }): void {
+
+        let advanceSearch = cloneDeep(this.advanceSearchRequest)
+        if (!advanceSearch.dataToSend.bsRangeValue) {
+            this.universalDate$.pipe(take(1)).subscribe(date => {
+                if (date) {
+                    advanceSearch.dataToSend.bsRangeValue = [moment(date[0], 'DD-MM-YYYY').toDate(), moment(date[1], 'DD-MM-YYYY').toDate()];
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:1822]: Use date format constant

File: apps/web-giddh/src/app/services/ledger.service.ts [hunk 5]
```diff
@@ -428,4 +436,37 @@ export class LedgerService {
...
+     * To get export Ledger's columnar report table
+     *
+     * @param {ReportsDetailedRequestFilter} model request model
+     * @param {string} companyUniqueName Company unique name
+     * @param {string} accountUniqueName Account unique name
+     * @param {*} [body] body
+     * @returns {Observable<BaseResponse<any, ReportsDetailedRequestFilter>>}
+     * @memberof LedgerService
+     */
+    public ExportLedgerColumnarReportTable(model: ReportsDetailedRequestFilter, companyUniqueName: string, accountUniqueName: string, body?: any): Observable<BaseResponse<any, ReportsDetailedRequestFilter>> {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/ledger.service.ts:450]: Use camel capsing for function name

File: apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.ts [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            if (response && response.status === 'success') {
+                if (response.body && response.body.results) {
+                    this.reportResponse = response.body;
+                    this.reportResponseResult = response.body.results;
+                    if (response.body.results.length) {
+                        this.prepareColumnForTable();
+                    }
+                }
+            } else {
+                this.toaster.errorToast(response.message);
```
[the remark this answers is not in the document]
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.ts:109]: Already done out side of block

File: apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.ts [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    /**
+     *  Life cycle hook for data-bound property of a directive changed
+     *
+     * @returns {void}
+     * @param {SimpleChanges} changes
+     * @memberof LedgerColumnarReportTableComponent
+     */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes && changes['columnarReportExportRequest'] && changes['columnarReportExportRequest'].currentValue) {
+            this.columnarReportExportRequest = changes['columnarReportExportRequest'].currentValue;
```
[the remark this answers is not in the document]
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.ts:69]: I know but It was not working some time, That's why have to

File: apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.html [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                    </tr>
+                </tbody>
+                <tbody *ngIf="reportResponse  &&  reportResponse?.totalItems === 0">
+                    <tr>
+                        <td [attr.colspan]="getTotalNoOfColumn()" align="center">No records found.</td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <div class="paginationWrapper" *ngIf="reportResponse && reportResponse?.totalItems > PAGINATION_LIMIT">
```
[the remark this answers is not in the document]
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/ledger-columnar-report-table/ledger.columnar.report.table.component.html:79]: No This is right we can use it after Import in ts

**Pull request merged**